### DISCUSSION TITLE
Test email throws timeout error

### DISCUSSION THREAD

**Discussion by LeqitDev** [pocketbase/pocketbase#4774] [Q&A]
Hey there,
lately trying to get PocketBase running for a website i made. Now i wanted to implement E-Mails to send verification, pw reset, etc emails. I tried to do that using my own mail server running on the same server both with docker. But on port 587 i get a cert error stating "unknown authority". On port 465 i get a timeout error same error i get wehn i use googles smtp.
My docker file looks like this:
```Dockerfile
FROM alpine:latest

ARG PB_VERSION=0.22.9
ARG TARGETPLATFORM

RUN if [ "$TARGETPLATFORM" = "linux/amd64" ]; then export ARCHITECTURE=amd64; \
    elif [ "$TARGETPLATFORM" = "linux/arm/v7" ]; then export ARCHITECTURE=arm; \
    elif [ "$TARGETPLATFORM" = "linux/arm64" ]; then export ARCHITECTURE=arm64; \
    else export ARCHITECTURE=amd64; fi; \
    echo "export ARCHITECTURE=${ARCHITECTURE}" >> /etc/profile

RUN apk add --no-cache \
    unzip \
    ca-certificates \
    wget && \
    update-ca-certificates

RUN source /etc/profile ; \
    cd /tmp ; \
    wget -O pb.zip https://github.com/pocketbase/pocketbase/releases/download/v${PB_VERSION}/pocketbase_${PB_VERSION}_linux_${ARCHITECTURE}.zip && \
    unzip /tmp/pb.zip -d /pb/ && \
    rm -f /tmp/pb.zip

EXPOSE 80

WORKDIR /pb

CMD ["/pb/pocketbase", "serve", "--http=0.0.0.0:80", "--dev"]
```
And this is my config in the mail settings:
![grafik](https://github.com/pocketbase/pocketbase/assets/34133543/fb868f64-d094-4f25-9910-744b14f69457)

Its all running behind a reverse proxy with this pocketbase instance on a custom location "api.cc-web.cloud/zerzinator"

Maybe somone has an idea what i can do :)

PS: the `--dev` option doesnt really show any helpful logs

**Comment by ganigeorgiev** [comment]
>  I tried to do that using my own mail server running on the same server both with docker.

Does it have a valid public certificate or you are using a self-signed one? If the latter then it may not work (at least not out of the box) because we don't expose the [`tls/Config.InsecureSkipVerify`](https://pkg.go.dev/crypto/tls#Config.InsecureSkipVerify) option and you may have to manually load/copy your certificate into the root certificates directory or (specify a custom one) (see https://github.com/pocketbase/pocketbase/discussions/4595#discussioncomment-8855032).

>  On port 465 i get a timeout error same error i get wehn i use googles smtp.

I'm not sure what could be the issue here but I suspect that it is something with the Gmail configuration. It is a little outdated but you could check https://github.com/pocketbase/pocketbase/discussions/458#discussioncomment-3651995. It is also possible that you may have to contact your hosting provider to allow 25/465/etc. outgoing ports.

**Reply by LeqitDev** [reply]
Thanks, for the fast reply and thanks for maintaining this really nice project. I use a self-signed certificate so i will try to load my certificate. As for the other point i already running another pb instance in a docker on the same machine and there everything works just fine with the same config (but another gmail account, pocketbase version and docker script) :/ I will reply my results after i tried it to load the certificate manually.

**Comment by cetaru** [comment]
I configured an application password in gmail,  and I keep getting the 'BadCredentials' error. It is running behind an NGINX with a Let's Encrypt SSL. Do I need to do any additional configuration?
I run the same crdentials on nodejs with nodemailder and it works fine.
